core/model/src/main/java/org/eclipse/rdf4j/model/Model.java

```diff
@@ -191,14 +191,14 @@ public default boolean remove(Resource subj, URI pred, Value obj, Resource... co
 	 *                  matching any one of these will match. To match statements without an associated context, specify
 	 *                  the value {@code null} and explicitly cast it to type {@code Resource}.
 	 * @return an {@link Iterator} over the statements in this Model that match the specified pattern.
-	 * 
+	 *
 	 * @since 3.2.0
-	 * 
+	 *
 	 * @see #filter(Resource, IRI, Value, Resource...)
 	 */
-	public default Iterator<Statement> getStatements(Resource subject, IRI predicate, Value object,
+	public default Iterable<Statement> getStatements(Resource subject, IRI predicate, Value object,
 			Resource... contexts) {
-		return filter(subject, predicate, object, contexts).iterator();
+		return () -> filter(subject, predicate, object, contexts).iterator();
```

Comment:
@jeenbroekstra I'm proposing we move from iterator to iterable because iterable supports for-each loops and is also simpler to convert into a stream.

Reply:
It's a pity that Java didn't introduce a combo interface with both iterable and stream. We can always create our own interface in the future though with extends both and still remain backwards compatible.

Reply:
> @jeenbroekstra I'm proposing we move from iterator to iterable because iterable supports for-each loops and is also simpler to convert into a stream.

I'm fine with that, but can you make sure you update the original issue's (#2030) description, and also the javadoc in Model?